trip_index.js

```diff
@@ -0,0 +1,158 @@
+  const id = event.target.className;
+  // making sure we clear out all trip deets
+  const tripDetails = $('#trip-deets');
+  tripDetails.empty();
+
+  // GET request, need to include the id of the
+  axios.get(tripURL + id)
+  .then((response) => {
+    reportStatus('Successfully loaded trip details');
+    // FIXME: Want to display the detail name before it by accessing the name of the keys
+    for (let detail1 in response.data) {
+      tripDetails.append(`<li> ${response.data[detail1]} </li>`);
+    }
+    // need to set the hidden value on the form with the corresponding trip ID
+    $('#tripID').val(id)
+  })
+  .catch((error) => {
+    console.log(error.response);
+    if (error.response && error.response.data.errors) {
+      reportError(
+        `Encountered an error: ${error.message}`,
+        error.response.data.errors
+      );
+    } else {
+      reportStatus(`Encountered an error: ${error.message}.You know what you did....`);
+    }
+  });
+};
+
+
+// read traveler's info from a form
+// $('#spot-form').serializeArray()
+const FORM_FIELDS = ['name', 'age', 'email'];
+const inputField = name => $(`#spot-form input[name="${name}"]`);
+
+const readFormData = () => {
+  const getInput = name => {
+    const input = inputField(name).val();
+    return input ? input : undefined;
+  };
+
+  const formData = {};
+  FORM_FIELDS.forEach((field) => {
+    formData[field] = getInput(field);
+  });
+  return formData;
+};
+
+// clear out a form to restock with new user info
+const clearForm = () => {
+  FORM_FIELDS.forEach((field) => {
+    inputField(field).val('');
+  });
+}
+
+
+//reserve a spot on a trip
+const holdSpot = (event) => {
+  event.preventDefault();
+  let spotId = $(`#tripID`).val();
+  const holdURL = `${tripURL}` + `${spotId}` + `/reservations`
+  // just for now, making sure that a spot hold can be created
+  // will replace with form input
+  const spotData = readFormData();
+  console.log(spotData);
+
+  reportStatus('Making sure we hold your spot...');
+
+  // POST request processing
+  axios.post(holdURL, spotData)
+
+  .then((response) => {
+    reportStatus(`Successfully reserved spot, enjoy your trip!`);
+    clearForm();
+  })
+  .catch((error) => {
+    console.log(error.response);
+    if (error.response && error.response.data.errors) {
+      reportError(
+        `Encountered an error: ${error.message}`,
+        error.response.data.errors
+      );
+    } else {
+      reportStatus(`Encountered an error: ${error.message}. You know what you did....`);
+    }
+  });
+};
+
+const reportStatus = (message) => {
+  $('#status-messages').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p>`
+  content += '<ul';
+  for (const field in errors) {
+    for (const problem in errors[field]) {
+      content += `<li>${field}: ${problem}</li>`
```

Comment:
Since `errors[field]` is an array, you probably want `for (const problem of errors[field])`. As is you get the indexes, not the strings.